### PR TITLE
New save and sample

### PR DESCRIPTION
**Pull request by CerebralSeed**
This currently is tested to work for both unconditional and upsampler. However, the coloring of the original images for the upsampler look off. I'm not sure what the issue is, so please have a look.

Note that this moves the generator kwargs arguments into a separate definition in the GigaGan class so it can be called in the save and sample, too.

### CODE REVIEW

**Comment by CerebralSeed** [comment]
Nevermind on the color issue. I was using some custom resizing with cv2 in the dataloader and forgot to convert the bgr to rgb after loading.

**Comment by CerebralSeed** [comment]
One other issue. It seems that just for the Generator, the image outputs are all identical on each individual save and sample. Upsampler doesn't have this problem.

**Comment by lucidrains** [comment]
@CerebralSeed thank you for getting the ball rolling here! that is good news that the upsampler works, as that unblocks Nicholas

let us merge this and i'll figure out what's wrong with the generator (should be really close to being functional!)

**Comment by CerebralSeed** [comment]
@lucidrains was retesting and it seems I messed up on the upsampler when copying over code or something. Easy fix though:
1. 
```
import torchvision # needed for the Resize after the interpolation, so the low res images are the same physical size as the upsampled image
```
2. Change line 2,000 to:
```
G_kwargs, maybe_text_kwargs, low_res_image = self.generate_kwargs(dl_iter, batch_size, sample=True)
```

**Comment by lucidrains** [comment]
@CerebralSeed try the updated version! i simplified some of the code, and also the base generator should be working now :partying_face: thank you for letting me know that it was only generating one image

**Comment by CerebralSeed** [comment]
It might have been that I set the learning rate too high and was overshooting. Retesting with a lower learning rate and higher gradient accumulation. Also, might try clipping the Generator loss, and see if that improves anything.
